cmake/Modules/HandleCUDAComputeCapability.cmake

```diff
@@ -8,24 +8,44 @@
 # those without the CUDA_ prefix
 
 if (NOT CUDA_TARGET_COMPUTE_CAPABILITY)
-	if("$ENV{CUDA_SM}" STREQUAL "")
-		set(ENV{CUDA_INCLUDE_DIRS} "${CUDA_INCLUDE_DIRS}")
-		set(ENV{CUDA_CUDART_LIBRARY} "${CUDA_CUDART_LIBRARY}")
-		set(ENV{CMAKE_CXX_COMPILER} "${CMAKE_CXX_COMPILER}")
-		execute_process(COMMAND bash -c "${CMAKE_CURRENT_SOURCE_DIR}/scripts/get_cuda_sm.sh"  OUTPUT_VARIABLE CUDA_TARGET_COMPUTE_CAPABILITY_) 
+	if(WIN32)
+		find_package(CUDA)
+		if (CUDA_FOUND)
+			set(CUDAFILE ${CMAKE_CURRENT_SOURCE_DIR}/examples/check_cuda.cu)
+			try_run(run_result compile_result ${CMAKE_CURRENT_BINARY_DIR} ${CUDAFILE}
+				RUN_OUTPUT_VARIABLE CUDA_TARGET_COMPUTE_CAPABILITY_
+				COMPILE_OUTPUT_VARIABLE compile_output)
+			if(NOT "${compile_result}")
+				message(SEND_ERROR "Detecting CUDA_TARGET_COMPUTE_CAPABILITY failed: Compilation failure")
+				message(SEND_ERROR "${compile_output}")
+			elseif(NOT ("${run_result}" EQUAL "0"))
+				message(SEND_ERROR "Detecting CUDA_TARGET_COMPUTE_CAPABILITY failed: Run failure")
+				message(SEND_ERROR "${CUDA_TARGET_COMPUTE_CAPABILITY_}")
+			endif()
+			separate_arguments(CUDA_TARGET_COMPUTE_CAPABILITY_ UNIX_COMMAND "${CUDA_TARGET_COMPUTE_CAPABILITY_}")
+			list(GET CUDA_TARGET_COMPUTE_CAPABILITY_ 1 FORMATTED_COMPUTE_CAPABILITY)
+			list(GET CUDA_TARGET_COMPUTE_CAPABILITY_ 0 CUDA_TARGET_COMPUTE_CAPABILITY_)
+		endif()
 	else()
-		set(CUDA_TARGET_COMPUTE_CAPABILITY_ $ENV{CUDA_SM})
-	endif()
+		if("$ENV{CUDA_SM}" STREQUAL "")
+			set(ENV{CUDA_INCLUDE_DIRS} "${CUDA_INCLUDE_DIRS}")
```

Comment:
Again, there are internal CMake variables for that.

Reply:
set(ENV stuff is from the current (unix) version of the module